edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/DeviceIdentityProvider.cs

```diff
@@ -49,10 +49,26 @@ public async Task<IDeviceIdentity> GetAsync(string clientId, string username, st
                 ClientInfo clientInfo = this.usernameParser.Parse(username);
                 clientInfo.ModelId.ForEach(async m => await this.metadataStore.SetModelId(clientInfo.DeviceId, m));
                 IClientCredentials deviceCredentials = null;
+                Option<IClientCredentials> actorCredentials = Option.None<IClientCredentials>();
 
                 if (!string.IsNullOrEmpty(password))
                 {
-                    deviceCredentials = this.clientCredentialsFactory.GetWithSasToken(clientInfo.DeviceId, clientInfo.ModuleId, clientInfo.DeviceClientType, password, false, clientInfo.ModelId, Option.None<string>());
+                    deviceCredentials = this.clientCredentialsFactory.GetWithSasToken(clientInfo.DeviceId, clientInfo.ModuleId, clientInfo.DeviceClientType, password, false, clientInfo.ModelId, clientInfo.AuthChain);
+
+                    Option<string> actorDeviceIdOption = AuthChainHelpers.GetActorDeviceId(clientInfo.AuthChain);
+                    actorDeviceIdOption.ForEach(actorDeviceId =>
```

Comment:
actorCredentials = actorDeviceIdOption.Map(a => this.clientCredentialsFactory.... )